RELEASE/scripts/autoscend.ash

```diff
@@ -2300,57 +2300,109 @@ boolean questOverride()
 	{
 		set_property("auto_wandOfNagamar", false);
 		return false;
 	}
-	else if (internalQuestStatus("questL13Final") == 12)
+	
+	if(creatable_amount($item[Wand Of Nagamar]) > 0)
+	{
+		return create(1, $item[Wand Of Nagamar]);
+	}
+	
+	//hunt for bear verb orgy
+	if (item_amount($item[Wand of Nagamar]) == 0 && internalQuestStatus("questL13Final") == 12)
 	{
 		return autoAdv($location[The VERY Unquiet Garves]);
 	}
-	else if(pulls_remaining() >= 2)
+	
+	//create items [WA] and [ND] if possible to make the pulls below be as optimal as possible
+	if(creatable_amount($item[WA]) > 0 && item_amount($item[WA]) == 0)
 	{
-		if((item_amount($item[ruby w]) > 0) && (item_amount($item[metallic A]) > 0))
-		{
-			cli_execute("make " + $item[WA]);
-		}
-		pullXWhenHaveY($item[WA], 1, 0);
-		pullXWhenHaveY($item[ND], 1, 0);
-		cli_execute("make " + $item[Wand Of Nagamar]);
-		return true;
+		create(1, $item[WA]);
 	}
-	else
+	if(creatable_amount($item[ND]) > 0 && item_amount($item[ND]) == 0)
 	{
-		if(auto_my_path() == "G-Lover")
+		create(1, $item[ND]);
+	}
+	
+	//these pulls can fail if you have not prepared the right items ahead of time. Trying all permutations to get missing parts
+	//if they do fail, you would have a pull remaining for clover
+	if (internalQuestStatus("questL13Final") == 11 && !in_hardcore())
+	{
+		if(pulls_remaining() >= 4 && item_amount($item[WA]) == 0 && item_amount($item[ND]) == 0)
 		{
-			pullXWhenHaveY($item[Ten-Leaf Clover], 1, 0);
+			pullXWhenHaveY($item[ruby W], 1, 0);
+			pullXWhenHaveY($item[metallic A], 1, 0);
+			pullXWhenHaveY($item[lowercase N], 1, 0);
+			pullXWhenHaveY($item[heavy D], 1, 0);
 		}
-		else
+		if(pulls_remaining() >= 2)
 		{
-			pullXWhenHaveY($item[Disassembled Clover], 1, 0);
+			pullXWhenHaveY($item[WA], 1, 0);
+			pullXWhenHaveY($item[ND], 1, 0);
+			if(item_amount($item[WA]) > 0 && item_amount($item[ND]) == 0)
+			{
+				pullXWhenHaveY($item[lowercase N], 1, 0);
+				pullXWhenHaveY($item[heavy D], 1, 0);
+			}
+			if(item_amount($item[WA]) == 0 && item_amount($item[ND]) > 0)
+			{
+				pullXWhenHaveY($item[ruby W], 1, 0);
+				pullXWhenHaveY($item[metallic A], 1, 0);
+			}
 		}
-		if(in_hardcore() && in_koe())
+		if(pulls_remaining() >= 1 && item_amount($item[WA]) > 0 && item_amount($item[ND]) == 0)
 		{
-			// TODO: Improve support
-			abort("In Kingdom of Exploathing: Please buy a Wand of Nagamar from the bazaar and re-run.");
-			return false;
+			pullXWhenHaveY($item[ND], 1, 0);
 		}
-		else if(cloversAvailable() > 0)
+		if(pulls_remaining() >= 1 && item_amount($item[WA]) == 0 && item_amount($item[ND]) > 0)
 		{
-			cloverUsageInit();
-			autoAdvBypass(322, $location[The Castle in the Clouds in the Sky (Basement)]);
-			cloverUsageFinish();
-			cli_execute("make " + $item[Wand Of Nagamar]);
-			return true;
+			pullXWhenHaveY($item[WA], 1, 0);
+		}
+	}
+	if(creatable_amount($item[Wand Of Nagamar]) > 0)
+	{
+		return create(1, $item[Wand Of Nagamar]);
+	}
```

Comment:
I would remove all this. Clovers are trivially cheap and most people have dozens in their inventory. Yes pulling the one item that you need to make a wand (usually the lowercase n in most runs) would save the adventure you spend clovering the castle basement but if we get to the point where we care about single adventure optimisations that require this much code, we have gone too far (and if users complain about it using an extra adventure, they shouldn't be running a script to ascend for them).

Reply:
change made on commit 1874d7f